sklearn/svm/_base.py

```diff
@@ -274,6 +274,16 @@ def fit(self, X, y, sample_weight=None):
             self.intercept_ *= -1
             self.dual_coef_ = -self.dual_coef_
 
+        # check for finiteness of coefficients
+        check_coef = (
+            np.isfinite(self._dual_coef_.data).all()
+            if self._sparse
+            else np.isfinite(self._dual_coef_).all()
+        )
+        if not (check_coef and np.isfinite(self._intercept_).all()):
+            # maybe check which coeff/feature is non-finite?
```

Comment:
Eventually, this can be split in two parts. I think it's fine to keep a single test.
```suggestion
        coef_finiteness = (
            np.isfinite(self._dual_coef_.data).all()
            if self._sparse
            else np.isfinite(self._dual_coef_).all()
        )
        intercept_finiteness = np.isfinite(self._intercept_).all()
        if not (coef_finiteness and intercept_finiteness):
```

Reply:
Thank you for your suggestion which I incorporated in the PR.